io/io/v7/src/TFile.cxx

```diff
@@ -191,7 +191,7 @@ std::string ROOT::Experimental::TFile::SetCacheDir(std::string_view path) {
   std::lock_guard<std::mutex> lock(GetCacheDirMutex());
 
   std::string ret = ::TFile::GetCacheFileDir();
-  ::TFile::SetCacheFileDir(path.to_string().c_str());
+  ::TFile::SetCacheFileDir(path.data());
```

Comment:
Actually, we don't need an overload just a replacement as a string_view can be constructed implicitly from a const char*.  I uploaded the change in signature of TFile::SetCacheFileDir.